gnomad_qc/v5/data_ingestion/federated_validity_checks.py

```diff
@@ -0,0 +1,50 @@
+"""Script to generate annotations for variant QC on gnomAD v4."""
+
+import argparse
+import logging
+
+import hail as hl
+
+from gnomad.assessment.validity_checks import summarize_variants
+from gnomad.resources.grch38.gnomad import public_release
+
+
+logging.basicConfig(format="%(levelname)s (%(name)s %(lineno)s): %(message)s")
+logger = logging.getLogger("federated_validity_checks")
+logger.setLevel(logging.INFO)
+
+
+def main(args):
+    """Perform validity checks for federated data."""
+    hl.init(
+        log="/federated_validity_checks.log",
+        tmp_dir="gs://gnomad-tmp-4day",
+    )
+    hl.default_reference("GRCh38")
+    test = args.test
+
+    # TODO: Add resources to intake federated data once obtained.
+    if test:
+        ht = public_release(data_type="exomes").ht()
+        ht = ht._filter_partitions(range(2))
```

Comment:
right now it's just to make sure things run (will be editing some functions to work better with Table format rather than a flattened info field), and we don't have any federated data to use. In the future the test option may be useful for federated data just to make sure nothing fails from some formatting issue. I can change the test function to allow a variable number of partitions to make it more flexible.